journalbeat/input/input.go

```diff
@@ -138,7 +138,9 @@ func (i *Input) publishAll(client beat.Client) {
 				select {
 				case <-i.done:
 					return
-				case v := <-c:
+				default:
+				}
+				for v := range c {
```

Comment:
our discussion this morning made me think and look at the code.
Instead of using the `default` route you can do the following, which replace 2 for with 1.

```golang
select {
case <-i.done:
	return
case v, ok := <-c: // ok is true if fetch is succesful.
	if !ok {
		return 
	}
    out <- v
}